codec-http/src/main/java/io/netty/handler/codec/http/CombinedHttpHeaders.java

```diff
@@ -36,8 +39,21 @@
  * Please refer to section <a href="https://tools.ietf.org/html/rfc7230#section-3.2.2">RFC 7230, 3.2.2</a>.
  */
 public class CombinedHttpHeaders extends DefaultHttpHeaders {
+    private static final Set<CharSequence> NON_COMBINABLE_HEADERS;
+
+    static {
+        NON_COMBINABLE_HEADERS = new HashSet<CharSequence>();
+
+        // According to the spec Set-cookie header does not use the list syntax and cannot be combined
+        NON_COMBINABLE_HEADERS.add(SET_COOKIE);
+    }
+
     public CombinedHttpHeaders(boolean validate) {
-        super(new CombinedHttpHeadersImpl(CASE_INSENSITIVE_HASHER, valueConverter(validate), nameValidator(validate)));
+        super(new CombinedHttpHeadersImpl(CASE_INSENSITIVE_HASHER, valueConverter(validate), nameValidator(validate), NON_COMBINABLE_HEADERS));
+    }
+
+    public CombinedHttpHeaders(boolean validate, Set<CharSequence> nonCombinableHeaders) {
```

Comment:
Adding this constructor seems like an overkill, to let users provide their own new headers-that-are-broken since [HTTP itself only specifies Set-Cookie as an exception](https://tools.ietf.org/html/rfc7230#section-3.2.2).